fpm/src/fpm.f90

```diff
@@ -154,8 +154,49 @@ subroutine build_model(model, settings, package, error)
     end do
     if (allocated(error)) return
 
+    ! Check for duplicate modules
+    call check_modules(model)
 end subroutine build_model
 
+! Check for duplicate modules
+subroutine check_modules(model) 
+    type(fpm_model_t), intent(in) :: model
+    integer :: maxsize
+    integer :: i,j,k,l,m,modi
+    type(string_t), allocatable :: modules(:)
+    ! Initialise the size of array
+    maxsize = 0
+    ! Get number of modules provided by each source file of every package
+    do i=1,size(model%packages)
+      do j=1,size(model%packages(i)%sources)
+        if (allocated(model%packages(i)%sources(j)%modules_provided)) then
+          maxsize = maxsize + size(model%packages(i)%sources(j)%modules_provided)
+        end if
+      end do
+    end do
+    ! Allocate array to contain distinct names of modules
+    allocate(modules(1:maxsize))
+
+    ! Initialise index to point at start of the newly allocated array
+    modi = 1
+
+    ! Loop through modules provided by each source file of every package
+    ! Add it to the array if it is not already there
+    ! Otherwise print out warning about duplicates
+    do k=1,size(model%packages)
+      do l=1,size(model%packages(k)%sources)
+        if (allocated(model%packages(k)%sources(l)%modules_provided)) then
+          do m=1,size(model%packages(k)%sources(l)%modules_provided)
+            if (model%packages(k)%sources(l)%modules_provided(m)%s.in.modules) then
+              print *,"Warning: Module ",model%packages(k)%sources(l)%modules_provided(m)%s," is duplicate"
```

Comment:
Let's print warnings to stderr instead of stdout, let's call them "Error" (because we can't proceed if there are duplicates). You will need to use `write(error_unit, *) ...`, where `error_unit` is a constant that comes from the intrinsic `iso_fortran_env` module.

Reply:
What does intrinsic mean btw? (I have implemented the requested change, but I have never seen the `intrinsic`.